Microsoft.Toolkit.Uwp.UI/Shadows/AttachedDropShadow.cs

```diff
@@ -0,0 +1,353 @@
+                if (mask == null && SupportsCompositionVisualSurface && CornerRadius > 0)
+                {
+                    // Create rounded rectangle geometry and add it to a shape
+                    var geometry = context.GetResource(RoundedRectangleGeometryResourceKey) ?? context.AddResource(
+                        RoundedRectangleGeometryResourceKey,
+                        context.Compositor.CreateRoundedRectangleGeometry());
+                    geometry.CornerRadius = new Vector2((float)CornerRadius);
+
+                    var shape = context.GetResource(ShapeResourceKey) ?? context.AddResource(ShapeResourceKey, context.Compositor.CreateSpriteShape(geometry));
+                    shape.FillBrush = context.Compositor.CreateColorBrush(Colors.Black);
+
+                    // Create a ShapeVisual so that our geometry can be rendered to a visual
+                    var shapeVisual = context.GetResource(ShapeVisualResourceKey) ??
+                                      context.AddResource(ShapeVisualResourceKey, context.Compositor.CreateShapeVisual());
+                    shapeVisual.Shapes.Add(shape);
+
+                    // Create a CompositionVisualSurface, which renders our ShapeVisual to a texture
+                    var visualSurface = context.GetResource(VisualSurfaceResourceKey) ??
+                                        context.AddResource(VisualSurfaceResourceKey, context.Compositor.CreateVisualSurface());
+                    visualSurface.SourceVisual = shapeVisual;
+
+                    // Create a CompositionSurfaceBrush to render our CompositionVisualSurface to a brush.
+                    // Now we have a rounded rectangle brush that can be used on as the mask for our shadow.
+                    var surfaceBrush = context.GetResource(SurfaceBrushResourceKey) ?? context.AddResource(
+                        SurfaceBrushResourceKey,
+                        context.Compositor.CreateSurfaceBrush(visualSurface));
+
+                    geometry.Size = visualSurface.SourceSize = shapeVisual.Size = context.Element.RenderSize.ToVector2();
+
+                    mask = surfaceBrush;
+                }
+            }
+
+            // Position our shadow in the correct spot to match the corresponding element.
+            context.SpriteVisual.Size = context.Element.RenderSize.ToVector2();
+            context.SpriteVisual.Offset = context.Element.CoordinatesFrom(CastTo).ToVector3();
+
+            return mask;
+        }
+
+        private void CustomMaskedElement_Loaded(object sender, RoutedEventArgs e)
+        {
+            var context = GetElementContext(sender as FrameworkElement);
+
+            context.Element.Loaded -= CustomMaskedElement_Loaded;
+
+            UpdateShadowClip(context);
+            UpdateShadowMask(context);
+        }
+
+        /// <inheritdoc/>
+        /*protected override CompositionClip GetShadowClip(AttachedShadowElementContext context)
```

Comment:
Oops, forgot to remove this, clipping doesn't apply here because we can't create a stroke based clip with Composition only compared to how `AttachedCardShadow` works.